buildSrc/src/main/java/org/elasticsearch/gradle/info/GlobalBuildInfoPlugin.java

```diff
@@ -213,26 +214,67 @@ private Object createGitRevisionResolver(final Project project) {
             @Override
             public String toString() {
```

Comment:
So this change doesn't actually remove any of the "laziness" introduced. Instead of returning an anonymous implementation of `Object` that overrides `toString()` the point here is to simply do this eagerly at config time, since it's quick. Or at least that was my understanding.

Reply:
Yes, my understanding from our discussion yesterday is that you would prefer this be kept lazy either way? My goal here is merely to make it faster.

Reply:
Ah ok, I though the main objection was to the complexity the laziness incurs. Let's do some tests of making this eager. If the config time penalty is negligible, let's rip that stuff out as I agree the complexity is only there to avoid the performance cost.

Reply:
Yes, I do not like the complexity that the laziness incurs. It would be great if the performance changes here from reading directly instead of forking are enough to support moving that out. I'll push a commit to remove the laziness, and you can test away. 😇